### PR TITLE
Introduce new APIs to read EC private keys and public keys

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
This PR is to introduce 4 new APIs to the Crypto module to interact with the private keys and public keys signed with "EC" algorithms.

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5821

## Examples

Newly introduced APIs,

Decode a 'EC' private key

```ballerina
crypto:PrivateKey privateKey = check crypto:decodeEcPrivateKeyFromKeyStore(keyStore, "keyAlias", "keyPassword");
```

Decode a 'EC' public key

```ballerina
crypto:PublicKey publicKey = check crypto:decodeEcPublicKeyFromTrustStore(trustStore, "keyAlias");
```

Generate a SHA384withECDSA based signature

```ballerina
byte[] signature = check crypto:signSha384withEcdsa(data, privateKey);
```

Verify a SHA384withECDSA based signature

```ballerina
boolean validity = check crypto:verifySha384withEcdsaSignature(data, signature, publicKey);
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java [hunk 1]
```diff
@@ -73,6 +74,26 @@ private Decode() {}
 
     public static Object decodeRsaPrivateKeyFromKeyStore(BMap<BString, BString> keyStoreRecord, BString keyAlias,
                                                          BString keyPassword) {
+        Object decodedPrivateKey = getPrivateKey(keyStoreRecord, keyAlias, keyPassword);
+        if (decodedPrivateKey instanceof PrivateKey) {
+            PrivateKey privateKey = (PrivateKey) decodedPrivateKey;
+            return buildRsPrivateKeyRecord(privateKey);
+        }
```
Comment (MohamedSabthar) [inline native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java:81]: I think with java17 we can do this
```suggestion
        if (decodedPrivateKey instanceof PrivateKey privateKey) {
            return buildRsPrivateKeyRecord(privateKey);
        }
```

Comment (MohamedSabthar) [reply native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java:81]: check other places

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java [hunk 2]
```diff
@@ -253,4 +320,8 @@ public static Object buildRsaPublicKey(BString modulus, BString exponent) {
             return CryptoUtils.createError("Algorithm of the key factory is not found: " + e.getMessage());
         }
     }
+
+    public static Object generateBCryptHash(BArray input, int cost) {
```
Comment (MohamedSabthar) [inline native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java:324]: Are we using this method anywhere? Are we not using the input arguments in method body?

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java [hunk 3]
```diff
@@ -133,28 +154,56 @@ public static Object decodeRsaPrivateKeyFromKeyFile(BString keyFilePath, Object
...
+        privateKeyRecord.put(StringUtils.fromString(Constants.PRIVATE_KEY_RECORD_ALGORITHM_FIELD),
+                             StringUtils.fromString(privateKey.getAlgorithm()));
+        return privateKeyRecord;
+    }
+
+    private static Object buildEcPrivateKeyRecord(PrivateKey privateKey) {
+        if (privateKey.getAlgorithm().equals(Constants.EC_ALGORITHM)) {
+            return getPrivateKeyRecord(privateKey);
+        } else {
+            return CryptoUtils.createError("Not a valid EC key");
```
Comment (MohamedSabthar) [inline native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java:186]: can directly return here without else block

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java (hunk 1 above)
Comment (Bhashinee) [reply native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java]: Fixed in [6f1b573](https://github.com/ballerina-platform/module-ballerina-crypto/pull/533/commits/6f1b5734ec97d4e3a5a4e254f44fe11bf72a1202)

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java (hunk 2 above)
Comment (Bhashinee) [reply native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java]: No, it is not being used. It was a mistake adding it to this PR. :) Removed in [6f1b573](https://github.com/ballerina-platform/module-ballerina-crypto/pull/533/commits/6f1b5734ec97d4e3a5a4e254f44fe11bf72a1202)

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java (hunk 3 above)
Comment (Bhashinee) [reply native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java]: Fixed in [6f1b573](https://github.com/ballerina-platform/module-ballerina-crypto/pull/533/commits/6f1b5734ec97d4e3a5a4e254f44fe11bf72a1202)

**Review by MohamedSabthar — COMMENTED** [review]
LGTM

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**